### ISSUE TITLE
Make NsqSharp .Net Core compatible

### ISSUE THREAD

**Issue by michiski** [judwhite/NsqSharp#84]
Hi, are there any plans to make your project .Net Core compatible?

**Comment by judwhite** [comment]
@michiski Yes, if it's possible without interfering too much with the regular .NET build. I still need to get up to speed on the latest with .NET Core. Last time I spoke with someone from Microsoft they were planning on adding support for `ServiceBase` but it wasn't ready yet; not sure how that's changed.

**Comment by tomc974** [comment]
Hi,
 
Are there any chance for this enhancement to be implemented in the coming months? I may have a need for an NSQ driver for a .NET Core 2.0 / .NET Standard 2.0 project.

As for `ServiceBase`, since this is a Windows-only feature, it probably never will be a part of neither .NET Core 2.0 nor .NET Standard 2.0.  My suggestion would be to split the NsqSharp project into two sub-projects (driver and bus) but this may be quite some work for you...

**Comment by matteocontrini** [comment]
`ServiceBase` is going to be made available (for Windows only) in .NET Core through the Windows Compatibility Pack, scheduled to be released together with .NET Core 2.1

https://blogs.msdn.microsoft.com/dotnet/2017/11/16/announcing-the-windows-compatibility-pack-for-net-core/

**Comment by judwhite** [comment]
@matteocontrini Thanks. Glad to see it's going to make it for Windows eventually. I'll probably end up doing something like https://github.com/judwhite/go-svc/blob/master/svc/svc_other.go and https://github.com/judwhite/go-svc/blob/master/svc/svc_windows.go with build tags to get a common interface across platforms, or use a plugin-like approach to achieve the same goal.

@crazybert has PR #91 started for .NET Core support.

**Comment by julealgon** [comment]
Hey @judwhite , are there any blockers to have a compilation target for .Net Core here still? We are facing a warning in one of our core projects due to this problem:

> Severity	Code	Description	Project	File	Line	Suppression State
Warning	NU1701	Package 'NsqSharp 0.5.4' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
